sdk/storage/azure_storage_blob/tests/blob_client.rs

```diff
@@ -58,3 +98,162 @@ async fn test_get_blob_properties_invalid_container() -> Result<(), Box<dyn Erro
+    );
+    assert_eq!(Bytes::from_static(data), response_body.collect().await?);
+
+    container_client.delete_container(None).await?;
+    Ok(())
+}
+
+#[recorded::test]
+async fn test_upload_blob(ctx: TestContext) -> Result<(), Box<dyn Error>> {
+    // Recording Setup
+    let recording = ctx.recording();
+    let mut options = BlobClientOptions::default();
+    recording.instrument(&mut options.client_options);
+
+    // Setup
+    let endpoint = format!(
+        "https://{}.blob.core.windows.net/",
+        recording.var("AZURE_STORAGE_ACCOUNT_NAME", None).as_str()
+    );
+
+    // Act
+    let container_client = BlobContainerClient::new(
+        &endpoint,
+        String::from("testcontainer3"),
+        recording.credential(),
+        Some(options.clone()),
+    )?;
+    container_client.create_container(None).await?;
+
+    let blob_client = BlobClient::new(
+        &endpoint,
+        String::from("testcontainer3"),
+        String::from("test_upload_blob.txt"),
+        recording.credential(),
+        Some(options),
+    )?;
+
+    let data = b"hello rusty world";
+    let response = blob_client
+        .upload_blob(
+            RequestContent::from(data.to_vec()),
+            false,
+            i64::try_from(data.len())?,
+            None,
+        )
+        .await?;
+
+    // Assert
+    assert_eq!(response.status(), StatusCode::Created);
+
+    container_client.delete_container(None).await?;
+    Ok(())
+}
+
+#[recorded::test]
+async fn test_upload_blob_overwrite(ctx: TestContext) -> Result<(), Box<dyn Error>> {
+    // Recording Setup
+    let recording = ctx.recording();
+    let mut options = BlobClientOptions::default();
+    recording.instrument(&mut options.client_options);
+
+    // Setup
+    let endpoint = format!(
+        "https://{}.blob.core.windows.net/",
+        recording.var("AZURE_STORAGE_ACCOUNT_NAME", None).as_str()
+    );
+
+    // Act
+    let container_client = BlobContainerClient::new(
+        &endpoint,
+        String::from("testcontainer4"),
+        recording.credential(),
+        Some(options.clone()),
+    )?;
+    container_client.create_container(None).await?;
+
+    let blob_client = BlobClient::new(
+        &endpoint,
+        String::from("testcontainer4"),
+        String::from("test_upload_blob_overwrite.txt"),
+        recording.credential(),
+        Some(options),
+    )?;
+
+    let data = b"hello rusty world";
+    blob_client
+        .upload_blob(
+            RequestContent::from(data.to_vec()),
+            false,
+            i64::try_from(data.len())?,
+            None,
+        )
+        .await?;
+
+    let data2 = b"hello overwritten rusty world";
+    let response = blob_client
+        .upload_blob(
+            RequestContent::from(data2.to_vec()),
+            true,
+            i64::try_from(data2.len())?,
+            None,
+        )
+        .await?;
+
+    // Assert
+    assert_eq!(response.status(), StatusCode::Created);
```

Comment:
You should check the content is indeed overwritten as expected.

Let's also test the error case (overwrite=false when there is a Blob). Can probably be in the same test right before we do overwrite=true

Reply:
Sounds good, I added in the error case, assert, and checking the content is in fact overwritten.